jolt-core/src/poly/commitment/bmmtv/poly_commit.rs

```diff
@@ -0,0 +1,411 @@
+        assert!(commitment_key_len >= bivariate_polynomial.y_polynomials.len());
+
+        let precomp_time = start_timer!(|| "Computing coefficients and KZG commitment");
+        let mut powers_of_x = vec![];
+        let mut cur = P::ScalarField::one();
+        for _ in 0..(commitment_key_len) {
+            powers_of_x.push(cur);
+            cur *= *x;
+        }
+
+        let coeffs = bivariate_polynomial
+            .y_polynomials
+            .iter()
+            .chain([UnivariatePolynomial::zero()].iter().cycle())
+            .take(commitment_key_len)
+            .map(|y_polynomial| {
+                let mut c = y_polynomial.coeffs.clone();
+                c.resize(kzg_srs.len(), <P::ScalarField>::zero());
+                c
+            })
+            .collect::<Vec<Vec<P::ScalarField>>>();
+        let y_eval_coeffs = (0..kzg_srs.len())
+            .map(|j| {
+                (0..commitment_key_len)
+                    .map(|i| powers_of_x[i] * coeffs[i][j])
+                    .sum()
+            })
+            .collect::<Vec<P::ScalarField>>();
+        // Can unwrap because y_eval_coeffs.len() is guarnateed to be equal to kzg_srs.len()
+        let y_eval_comm = P::G1::msm(kzg_srs.g1_powers(), &y_eval_coeffs).unwrap();
+        end_timer!(precomp_time);
+
+        let ipa_time = start_timer!(|| "Computing IPA proof");
+        let ip_proof = MippK::<P, ProofTranscript>::prove(
+            kzg_srs,
+            (y_polynomial_comms, powers_of_x),
+            transcript,
+        )?;
+        end_timer!(ipa_time);
+        let kzg_time = start_timer!(|| "Computing KZG opening proof");
+        let (kzg_proof, _eval) =
+            UnivariateKZG::<P>::open(kzg_srs, &UnivariatePolynomial::from_coeff(y_eval_coeffs), y)?;
+        end_timer!(kzg_time);
+
+        Ok(OpeningProof {
+            ip_proof,
+            y_eval_comm,
+            kzg_proof: kzg_proof.into_group(),
+        })
+    }
+
+    pub fn verify(
+        v_srs: &KZGVerifierKey<P>,
+        com: PairingOutput<P>,
+        point: (P::ScalarField, P::ScalarField),
+        eval: P::ScalarField,
+        proof: &OpeningProof<P>,
+        transcript: &mut ProofTranscript,
+    ) -> Result<bool, Error> {
+        let (x, y) = point;
+        let ip_proof_valid = MippK::<P, ProofTranscript>::verify(
+            v_srs,
+            (com, proof.y_eval_comm),
+            x,
+            &proof.ip_proof,
+            transcript,
+        )?;
+        let kzg_proof_valid = UnivariateKZG::<P>::verify(
+            v_srs,
+            &proof.y_eval_comm.into_affine(),
+            &y,
+            &proof.kzg_proof.into_affine(),
+            &eval,
+        )?;
+        println!(
+            "ip_proof_valid: {}, kzg_proof_valid: {}",
+            ip_proof_valid, kzg_proof_valid
+        );
+        Ok(ip_proof_valid && kzg_proof_valid)
+    }
+}
+
+pub struct UnivariatePolynomialCommitment<P, D> {
+    _pairing: PhantomData<P>,
+    _digest: PhantomData<D>,
+}
```

Comment:
For symmetry with BivariatePolynomialCommitment:
```suggestion
pub struct UnivariatePolynomialCommitment<P: Pairing, D>(MippK<P, D>);
```